### DISCUSSION TITLE
Questions for nvim-treesitter team from a plugin developer

### DISCUSSION THREAD

**Discussion by RRethy** [nvim-treesitter/nvim-treesitter#5379] [General]
I don't see where this is documented other than an [PR](https://github.com/RRethy/vim-illuminate/pull/170) that was opened in one of my repos, but there seems to be changes coming that will eventually break backwards compatibility. I have a few questions I can't find the answer to.

- Is this documented anywhere?
- If we can't use `define_modules`, how do we check if treesitter is attached to the buffer?
- How should we be checking if query files are present, e.g. `has_query_files` and `has_locals` were both provided by `nvim-treesitter.query` previously, but I don't see that on `main`?

**Comment by clason** [comment]
1. https://github.com/nvim-treesitter/nvim-treesitter/issues/4767
2. you don't need to; you can just run whatever you want. ("treesitter" doesn't attach; that was a thing invented for modules. Check the `indent.lua` file to see how to turn a module into a standalone thing.)
3. `vim.treesitter.query.get(lang, type)`
